### ISSUE TITLE
CTC loss turns to be negative in some training epochs of ASR in ESPnet1

### ISSUE THREAD

**Issue by Khelp** [espnet/espnet#2669]
Dear Everyone,

I have trained a Transformer-based ASR model by using my own dataset.
I usually get the negative loss in some epochs:
`{
        "main/loss_ctc": -5.671372443975481e+35,
        "main/loss_att": 15.803862244486808,
        "main/acc": 0.9607279838545754,
        "main/loss": -1.1342744887950961e+35,
        "epoch": 9,
        "iteration": 122600,
        "elapsed_time": 28301.330499992007
    },`
`{
        "main/loss_ctc": -5.671372443975481e+35,
        "main/loss_att": 15.180414428114892,
        "main/acc": 0.9647341349090253,
        "main/loss": -1.1342744887950961e+35,
        "epoch": 17,
        "iteration": 234600,
        "elapsed_time": 50239.392387389904
    },`

I have checked my dataset, there is no blank label in all of my training data.
Could someone tell me:
1. How could it happens?
2. How to find and solve this problem?
3. is it affected to the ASR performance?

Thanks in advance!

**Comment by sw005320** [comment]
> 1. How could it happens?

It happens when the output length is relatively longer than subsampled input length.
It mainly comes from the irregular files (original recordings or transcriptions are wrong).

> 2. How to find and solve this problem?

I recommend you to remove such utterances.
You may use
https://github.com/espnet/espnet/blob/master/egs/wsj/asr1/run.sh#L146-L155

> 3. is it affected to the ASR performance?

It would affect performance.
So, I recommend you to remove it.

It would be great if you identify which file shows such issues and report the input and output lengths of these files.
We may make a script to automatically identify these files and remove them in the data preparation stage.

**Comment by Khelp** [comment]
Dear @sw005320,

Thanks for your comments and useful solutions.
I have found 4 utterances that have a problem in its' transcription:
utt1 (input: 802, output:2815)
utt2 (input: 2733, output:3282)
utt3 (input: 1328, output:1548)
utt4 (input: 1006, output:1602)

Best regards,

**Comment by sw005320** [comment]
I see.
Thanks!
This is very informative.

Year, it seems that the output length is extremely large.
In addition to this, due to efficient computation, we usually remove very long input/output utterances.
https://github.com/espnet/espnet/blob/master/egs/librispeech/asr1/run.sh#L116-L117
The above case can be simply removed with these steps.

**Comment by Khelp** [comment]
Thanks for your value comments. I will take a note on it.
